### PR TITLE
(React) dynamic-list-of-todos

### PR DESCRIPTION
**Pull request by Ostafii4uk**
[DEMO LINK](https://ostafii4uk.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,25 +1,44 @@
...
-  ] = useState(0);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [todos, setTodos] = useState([]);
+  const [users, setUsers] = useState([]);
+
+  useEffect(() => {
+    getUsers()
+      .then(usersFromServer => {
+        setUsers(usersFromServer);
+      });
```
Comment (jstmpelowycz) [inline src/App.tsx:17]: You might have missed one of the requirements in technical task:
*User component receives userId as a prop and loads user details from [GET user endpoint](https://mate.academy/students-api/users/1) (replace 1 with a given userId)*

File: src/App.tsx [hunk 2]
```diff
@@ -1,25 +1,44 @@
...
+          onSelectedUserId={setSelectedUserId}
+        />
       </div>
 
       <div className="App__content">
         <div className="App__content-container">
           {selectedUserId ? (
-            <CurrentUser />
+            <CurrentUser
+              users={users}
```
Comment (jstmpelowycz) [inline src/App.tsx:38]: In accordance to the comment above, you shouldn't have passed `users` to `User`, two down below only

File: src/components/CurrentUser/CurrentUser.tsx [hunk 3]
```diff
@@ -1,12 +1,49 @@
...
+  // eslint-disable-next-line no-console
+  console.log(result);
+
+  return (
+    <div className="CurrentUser">
+      <h2 className="CurrentUser__title">
+        <span>
+          Selected user:
+          {' '}
+          {result?.id}
```
Comment (jstmpelowycz) [inline src/components/CurrentUser/CurrentUser.tsx:31]:
```suggestion
          {`Selected user: ${result?.id}`}
```

File: src/components/TodoList/TodoList.tsx [hunk 4]
```diff
@@ -1,44 +1,106 @@
...
+      .filter(todo => {
+        switch (selectedActivities) {
+          case 'active':
+            return !todo.completed;
+          case 'completed':
+            return todo.completed;
+          case 'select all':
+            return todo;
+          default:
+            return todo;
```
Comment (jstmpelowycz) [inline src/components/TodoList/TodoList.tsx:31]: I would recommend using enum

File: src/components/TodoList/TodoList.tsx [hunk 5]
```diff
@@ -1,44 +1,106 @@
...
+          </option>
+          <option>
+            active
+          </option>
+          <option>
+            completed
+          </option>
+          <option>
+            select all
+          </option>
```
Comment (jstmpelowycz) [inline src/components/TodoList/TodoList.tsx:69]: Simplify your own code, read [this](https://stackoverflow.com/questions/55601342/using-enumerations-in-react-select)

File: src/components/TodoList/TodoList.tsx [hunk 6]
```diff
@@ -1,44 +1,106 @@
...
+        </select>
+        <ul className="TodoList__list">
+          {filterTodos.map(todo => (
+            <li
+              key={todo.id}
+              className={classnames('TodoList__item', (todo.completed
+                ? 'TodoList__item--checked' : 'TodoList__item--unchecked'
+              ))}
```
Comment (jstmpelowycz) [inline src/components/TodoList/TodoList.tsx:77]:
```suggestion
              className={classnames('TodoList__item', (todo.completed
                  ? 'TodoList__item--checked' 
                  : 'TodoList__item--unchecked'
              ))}
```

File: src/react-app-env.d.ts [hunk 7]
```diff
@@ -1,3 +1,20 @@
 /// <reference types="react-scripts" />
 
-type Todo = {}; /* TODO: DESCRIBE */
+export type Todo = {
+  id: number,
+  createdAt?: string,
+  updatedAt?: string,
```
Comment (jstmpelowycz) [inline src/react-app-env.d.ts:6]: These are not necessary here and down below in `User` type

**Review by jstmpelowycz — CHANGES_REQUESTED** [review]
A few things to reflect upon and commit changes, good job though!!

**Review by Ihor-Karpyn — APPROVED** [review]

**Pull request closed without merging**